File: Stage12/Python/Problem1018.py
def main():
    N, M = map(int, input().split())
    board = [input().strip() for _ in range(N)]

    def repaint_count(x, y):
        cnt1, cnt2 = 0, 0
        for i in range(8):
            for j in range(8):
                current = board[x+i][y+j]
                if (i + j) % 2 == 0:
                    if current != 'W':
                        cnt1 += 1
                    if current != 'B':
                        cnt2 += 1
                else:
                    if current != 'B':
                        cnt1 += 1
                    if current != 'W':
                        cnt2 += 1
        return min(cnt1, cnt2)

    result = float('inf')
    for i in range(N - 7):
        for j in range(M - 7):
            result = min(result, repaint_count(i, j))

    print(result)

if __name__ == "__main__":
    main()

File: Stage12/Java/Problem1018.java
package Stage12.Java;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Problem1018 {
    static char[][] board;
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        int N = Integer.parseInt(st.nextToken());
        int M = Integer.parseInt(st.nextToken());
    
        board = new char[N][M];
        for (int i = 0; i < N; i++) {
            String line = br.readLine();
            for (int j = 0; j < M; j++) {
                board[i][j] = line.charAt(j);
            }
        }

        int result = Integer.MAX_VALUE;
        for (int i = 0; i <= N - 8; i++) {
            for (int j = 0; j <= M - 8; j++) {
                result = Math.min(result, repaintCount(i, j));
            }
        }

        System.out.println(result);
        br.close();
    }

    private static int repaintCount(int x, int y) {
        int cnt1 = 0;
        int cnt2 = 0;

        for (int i = 0; i < 8; i++) {
            for (int j = 0; j < 8; j++) {
                char current = board[x + i][y + j];

                if ((i + j) % 2 == 0) {
                    if (current != 'W') cnt1++;
                    if (current != 'B') cnt2++;
                } else {
                    if (current != 'B') cnt1++;
                    if (current != 'W') cnt2++;
                }
            }
        }

        return Math.min(cnt1, cnt2);
    }
}
